.github/workflows/prepare-bundled-themes.yml

```diff
@@ -0,0 +1,157 @@
+
+jobs:
+  # Prepares bundled themes for release.
+  #
+  # Performs the following steps:
+  # - Checks out the repository.
+  # - Installs Node.js (only when theme has a build process)
+  # - Installs npm dependencies (only when theme has a build process)
+  # - Runs the theme build script (only when theme has a build process)
+  # - Ensures version-controlled files are not modified or deleted.
+  # - Creates a ZIP file of the theme for wordpress.org.
+  # - Uploads the ZIP file as a workflow artifact.
+  bundle-theme:
+    name: Prepare ${{ matrix.theme }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    if: ${{ github.repository == 'WordPress/wordpress-develop' || github.event_name == 'pull_request' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        theme: [
+            'twentytwentythree',
+            'twentytwentytwo',
+            'twentytwentyone',
+            'twentytwenty',
+            'twentynineteen',
+            'twentyseventeen',
+            'twentysixteen',
+            'twentyfifteen',
+            'twentyfourteen',
+            'twentythirteen',
+            'twentytwelve',
+            'twentyeleven',
+            'twentyten'
+        ]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
+        with:
+          ref: ${{ github.event_name == 'workflow_dispatch' && inputs.branch || github.ref }}
+
+      - name: Install Node.js
+        uses: actions/setup-node@8c91899e586c5b171469028077307d293428b516 # v3.5.1
+        if: ${{ contains( fromJson( '["twentynineteen", "twentytwenty", "twentytwentyone"]' ), matrix.theme ) }}
+        with:
+          node-version-file: '.nvmrc'
+          cache: npm
+          cache-dependency-path: src/wp-content/themes/${{ matrix.theme }}/package-lock.json
+
+      - name: Install npm dependencies
+        if: ${{ contains( fromJson( '["twentynineteen", "twentytwenty", "twentytwentyone"]' ), matrix.theme ) }}
+        run: npm ci
+        working-directory: src/wp-content/themes/${{ matrix.theme }}
+
+      - name: Build theme
+        if: ${{ contains( fromJson( '["twentynineteen", "twentytwenty", "twentytwentyone"]' ), matrix.theme ) }}
+        run: npm run build
+        working-directory: src/wp-content/themes/${{ matrix.theme }}
+
+      - name: Ensure version-controlled files are not modified or deleted
+        if: ${{ contains( fromJson( '["twentynineteen", "twentytwenty", "twentytwentyone"]' ), matrix.theme ) }}
+        run: git diff --exit-code
+        working-directory: src/wp-content/themes/${{ matrix.theme }}
+
+      - name: Create theme ZIP file
+        run: zip -r ${{ matrix.theme }}.zip src/wp-content/themes/${{ matrix.theme }}
+
+      - name: Upload theme ZIP as an artifact
+        uses: actions/upload-artifact@83fd05a356d7e2593de66fc9913b3002723633cb # v3.1.1
+        with:
+          name: ${{ matrix.theme }}
+          path: ${{ matrix.theme }}.zip
```

Comment:
Once this is closer to done, it might be worth setting `retention-days: <shorter than the default>` as some of the files are quite large and they probably don't need to stick around for too long.

It might be that the files are large due to `node_modules` so it's probably worth rechecking once the zips are cleaned of unneeded files.

Reply:
Excluding `node_modules` made a considerable difference (from ~50MB to less than 1MB). With his in mind, I don't think we need to shorten retention for storage purposes. But the intent of the workflow is to generate the ZIP files for a release after confirming the overall build script works as intended.

Let's try using a week to start and we can change it if we find that's not long enough.